src/main/java/com/cloudbees/jenkins/plugins/bitbucket/hooks/WebhookCause.java

```diff
@@ -0,0 +1,23 @@
+package com.cloudbees.jenkins.plugins.bitbucket.hooks;
+
+import hudson.model.Cause;
+import org.kohsuke.stapler.export.Exported;
+
+public class WebhookCause extends Cause {
+    private String payload;
+
+    public WebhookCause(String payload) {
+        this.payload = payload;
```

Comment:
the entire payload it's too much (maybe it's not safe neither to expose)
Store only the type of webhook event so in the cause you will see something like "pr::created event"